loadbalancer/etcd/publisher_test.go

```diff
@@ -49,6 +50,8 @@ func TestPublisher(t *testing.T) {
 	if _, err := p.Endpoints(); err != nil {
 		t.Fatal(err)
 	}
+	// wait for the loop before stop
+	time.Sleep(time.Millisecond)
```

Comment:
From my current understanding https://github.com/go-kit/kit/blob/master/loadbalancer/etcd/publisher_test.go#L47 stops the loop before the mock

```
func (c *fakeClient) WatchPrefix(prefix string, responseChan chan *stdetcd.Response) {
	response, ok := c.responses[prefix]
	if ok {
		responseChan <- response
	}
}
```
sends to response channel. Looks like a hack, isn't it?